theano/tensor/opt.py

```diff
@@ -3421,7 +3423,26 @@ def local_fill_sink(node):
     c = node.op(*inputs)
     for model in models:
         c = T.fill(model, c)
-    return [c]
+
+    # The newly created node c doesn't has 'clients',
+    # so this iteration is took place with node.outputs[0]
+    replacements = {node.outputs[0]: c}
+    for client, cl_idx in node.outputs[0].clients:
+        if (hasattr(client, 'op') and
+                isinstance(client.op, T.Elemwise) and
+                not client.op == T.fill):
+            client_inputs = client.inputs[:]
+            client_inputs[cl_idx] = c
+            new_client = client.op(*client_inputs)
+
+            # Add clients to new_client
+            new_client.owner.outputs[0].clients = client.outputs[0].clients
+            r = local_fill_sink.transform(new_client.owner)
+            if r is False:
+                continue
+            replacements.pop(node.outputs[0], None)
```

Comment:
Besides, if `node.outputs[0]` is included in `replacements`, is it possible that there will be two versions after the replacement happen?